### PR TITLE
Add ability to configure auth backends and classes

### PR DESCRIPTION
**Pull request by Scnaeg**

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
It would be nice to have some tests for this.

**Comment by Scnaeg** [comment]
> It would be nice to have some tests for this.

I've added some tests. Unfortunately bundler has conflicting requirements on the RHEL 8 system I have available so I couldn't actually run the tests and verify the tests.

File: spec/acceptance/settings_spec.rb [hunk 1]
```diff
@@ -87,6 +87,95 @@ class { 'pulpcore':
...
+            'pulpcore.app.authentication.PulpNoCreateRemoteUserBackend',
+          ],
+        }
+        PUPPET
+      end
+    end
+
+    describe file('/etc/pulp/settings.py') do
+      it { is_expected.to be_file }
+      its(:content) { is_expected.to include('AUTHENTICATION_BACKENDS = ["django.contrib.auth.backends.ModelBackend", "pulpcore.app.authentication.PulpNoCreateRemoteUserBackend"]$') }
```
Comment (ekohl) [inline spec/acceptance/settings_spec.rb:122]: This looks like a regex leftover
```suggestion
      its(:content) { is_expected.to include('AUTHENTICATION_BACKENDS = ["django.contrib.auth.backends.ModelBackend", "pulpcore.app.authentication.PulpNoCreateRemoteUserBackend"]') }
```

File: spec/acceptance/settings_spec.rb [hunk 2]
```diff
@@ -87,6 +87,95 @@ class { 'pulpcore':
...
+
+  context 'REST_FRAMEWORK__DEFAULT_AUTHENTICATION_CLASSES set' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        class { 'pulpcore':
+          rest_framework_default_authentication_classes => [
+            'rest_framework.authentication.BasicAuthentication',
+            'rest_framework.authentication.SessionAuthentication',
+            'pulpcore.app.authentication.PulpRemoteUserAuthentication'
```
Comment (ekohl) [inline spec/acceptance/settings_spec.rb:158]: Wouldn't we expect a trailing comma here?
```suggestion
            'pulpcore.app.authentication.PulpRemoteUserAuthentication',
```

File: spec/acceptance/settings_spec.rb [hunk 3]
```diff
@@ -87,6 +87,95 @@ class { 'pulpcore':
...
+      end
+    end
+
+    describe file('/etc/pulp/settings.py') do
+      it { is_expected.to be_file }
+        is_expected.to include <<~EXPECTED
+          REST_FRAMEWORK__DEFAULT_AUTHENTICATION_CLASSES = (
+              'rest_framework.authentication.BasicAuthentication',
+              'rest_framework.authentication.SessionAuthentication',
+              'pulpcore.app.authentication.PulpRemoteUserAuthentication'
```
Comment (ekohl) [inline spec/acceptance/settings_spec.rb:171]: And here
```suggestion
              'pulpcore.app.authentication.PulpRemoteUserAuthentication',
```

**Comment by Scnaeg** [comment]
Thank you very much for the review and I'm so sorry for the leftovers. Definitely not my day...

**Review by ekohl — APPROVED** [review]

**Comment by Odilhao** [comment]
The tests for 3.39, 3.49 and nightly failed after this merge, not sure if it's related.

https://community.theforeman.org/t/pulpcore-nightly-rpm-pipeline-395-failed/39058

**Comment by evgeni** [comment]
Well, first of all the CI was broken due to a syntax error: https://github.com/theforeman/puppet-pulpcore/pull/349

**Comment by ekohl** [comment]
This isn't the first time auto-merge ignores a required check. At this point I really don't trust GH's auto-merge anymore.

**Pull request merged**